smm/lotus_adapter.go

```diff
@@ -0,0 +1,253 @@
+        Value:    types.NewInt(0),
+        GasLimit: types.NewInt(1000000),
+        GasPrice: types.NewInt(1),
+    }
+    signedMsg, err := adapter.fullNode.MpoolPushMessage(ctx, &msg)
+    if err != nil {
+        return cid.Undef, err
+    }
+    return signedMsg.Cid(), nil
+}
+
+func (adapter lotusAdapter) Start() {
+    go adapter.eventHandler()
+}
+
+func (adapter lotusAdapter) MostRecentState(ctx context.Context) (StateKey, error) {
+    ts, err := adapter.fullNode.ChainHead(ctx)
+    if err != nil {
+        return "", err
+    }
+    tsk := types.NewTipSetKey(ts.Cids()...)
+    return StateKey(tsk.Bytes()), nil
+}
+
+func (adapter lotusAdapter) GetMinerState(ctx context.Context, stateKey StateKey) (*MinerChainState, error) {
+    state := new(MinerChainState)
+    ts, err := statekey2tipset(ctx, stateKey, adapter.fullNode)
+    if err != nil {
+        return nil, err
+    }
+    sectors, err := adapter.fullNode.StateMinerSectors(ctx, adapter.miner, ts)
+    if err != nil {
+        return nil, err
+    }
+    provingSectors, err := adapter.fullNode.StateMinerProvingSet(ctx, adapter.miner, ts)
+    if err != nil {
+        return nil, err
+    }
+    state.Sectors = make(map[uint64]api.ChainSectorInfo)
+    for _, sectorInfo := range sectors {
+        state.Sectors[sectorInfo.SectorID] = *sectorInfo
+    }
+    for _, sectorInfo := range provingSectors {
+        state.ProvingSet[sectorInfo.SectorID] = struct{}{}
+    }
+    state.Address = Address(adapter.miner.String())
+    state.PreCommittedSectors = make(map[uint64]api.ChainSectorInfo)
+    state.StagedCommittedSectors = make(map[uint64]api.ChainSectorInfo)
+    state.ProvingSet = make(map[uint64]struct{})
+    return state, nil
+}
+
+func (adapter lotusAdapter) GetRandomness(ctx context.Context, stateKey StateKey, offset uint) ([]byte, error) {
+    ts, err := statekey2tipset(ctx, stateKey, adapter.fullNode)
+    if err != nil {
+        return nil, err
+    }
+    tsk := types.NewTipSetKey(ts.Cids()...)
+    return adapter.fullNode.ChainGetRandomness(ctx, tsk, ts.Blocks()[0].Tickets, int(offset))
+}
+
+func (adapter lotusAdapter) GetProvingPeriod(ctx context.Context, stateKey StateKey) (*ProvingPeriod, error) {
+    pp := new(ProvingPeriod)
+    ts, err := statekey2tipset(ctx, stateKey, adapter.fullNode)
+    if err != nil {
+        return nil, err
+    }
+    end, err := adapter.fullNode.StateMinerProvingPeriodEnd(ctx, adapter.miner, ts)
+    if err != nil {
+        return nil, err
+    }
+    pp.End = Epoch(end)
+    if end > build.ProvingPeriodDuration {
+        pp.Start = Epoch(end - build.ProvingPeriodDuration)
+    } else {
+        pp.Start = 0
+    }
+
+    return pp, nil
+}
+
+func (adapter lotusAdapter) SubmitSelfDeals(ctx context.Context, deals []uint64) (cid.Cid, error) {
+    // TODO: Adapt from shared modules' definition of storage deals to lotus' storage deal
+    // TODO: Each deal has to be signed (adapter.fullNode.WalletSign)
```

Comment:
If the prerequisites aren't ready for this I suggest leaving it out of this PR rather than committing it broken.